### ISSUE TITLE
make relevance limit configurable

### ISSUE THREAD

**Issue by twollnik** [hpi-swa-teaching/InteractiveProfilingTool#31]
Als Squeak Entwickler*in möchte ich einstellen können, ab wie viel Prozent der Gesamtlaufzeit mir methoden angezeigt werden, um wenig relevante Methoden ausfiltern zu können.

**Comment by linusha** [comment]
Estimate: 1h

**Comment by tillprochaska** [comment]
I think this is related to (or even a duplicate of?) #7. I think there are at least to relevant, open questions in order to estimate the cost of this story:

* Should methods/subtress be hidden completely or should the threshold for visual highlighting be customizable?
* How should users be able to customize the threshold? Using a UI / input field? Should this setting be persisted?

Also, I think it’d be sensible to include not only relative, but also absolute runtime.

**Comment by linusha** [comment]
@tillprochaska no, this is not a duplicate. The question raised by you belong to #7. 
This Issue is about a relevance threshhold in the getChildren part of the ```IPTTree``` where currently the value is always 1. This means that only nodes with a %runtime => 1% of the overall runtime are displayed in the resulting tree. This is because otherwise the results became extremely large and messy, according to @Alexander-Dubrawski.
After talking with Eva on Monday she agreed that this is sensible but that the threshhold should be changeable. 
TLDR: This is about the containment in the tree at all while #7 is about highlighting something in the tree.

However, your are right that this requires some thinking about UI and work with the ToolBuilder, so I upped my estimate accordingly.

**Comment by twollnik** [comment]
Estimate: 3h

**Comment by Alexander-Dubrawski** [comment]
Estimate: 3h

**Comment by linusha** [comment]
UI research @nickpodratz

**Comment by tillprochaska** [comment]
In Squeak, in order to allow users to customize certain settings, there is the PreferenceBrowser which is an interface to the Preferences class in the System-Preferences package. That class is extensible, so we could simply add a new preference for e. g. the threshold. The class comment for Preferences is quite good.

**Comment by nickpodratz** [comment]
Eva said that the preference browser is more aimed for system wide configuration, less for parameterisation procedures. We should therefore integrate this rather into our own window.
